modules/libarchive/3.7.3/patches/build_with_bazel.patch

```diff
@@ -0,0 +1,1805 @@
+diff --git a/WORKSPACE.bzlmod b/WORKSPACE.bzlmod
+new file mode 100644
+index 00000000..8b137891
+--- /dev/null
++++ b/WORKSPACE.bzlmod
+@@ -0,0 +1 @@
++
+diff --git a/bazel/BUILD.bazel b/bazel/BUILD.bazel
+new file mode 100644
+index 00000000..e1e1a513
+--- /dev/null
++++ b/bazel/BUILD.bazel
+@@ -0,0 +1 @@
++# empty for patch sake
+diff --git a/bazel/config.bzl b/bazel/config.bzl
+new file mode 100644
+index 00000000..1519aedb
+--- /dev/null
++++ b/bazel/config.bzl
+@@ -0,0 +1,8 @@
++PLATFORM_LOCAL_DEFINES = select({
++    "@platforms//os:windows": [
++        "PLATFORM_CONFIG_H=\\\"libarchive_bazel_windows_config.h\\\"",
++    ],
++    "//conditions:default": [
++        "PLATFORM_CONFIG_H=\\\"libarchive_bazel_generic_config.h\\\"",
++    ],
++})
+diff --git a/bazel/libarchive_test.bzl b/bazel/libarchive_test.bzl
+new file mode 100644
+index 00000000..b7744dee
+--- /dev/null
++++ b/bazel/libarchive_test.bzl
+@@ -0,0 +1,31 @@
++load("@bazel_skylib//rules:write_file.bzl", "write_file")
++load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
++
++def libarchive_test(name = None, tests = None, data = None, deps = [], **kwargs):
++    if not tests:
++        tests = [name]
++    if not data:
++        data = native.glob(["*.uu"])
++    write_file(
++        name = "{}_list_h".format(name),
++        out = "_{}_list_h/list.h".format(name),
++        content = ["DEFINE_TEST({})".format(test) for test in tests],
++    )
++    cc_library(
++        name = "{}_list".format(name),
++        hdrs = ["_{}_list_h/list.h".format(name)],
++        strip_include_prefix = "_{}_list_h".format(name),
++    )
++    cc_test(
++        name = name,
++        includes = ["."],
++        srcs = ["//test_utils:test_main.c"] + ["{}.c".format(name)],
++        data = data,
++        deps = deps + [
++            ":{}_list".format(name),
++            "//test_utils:common",
++            "//libarchive",
++            "//libarchive:internal_hdrs",
++        ],
++        **kwargs
++    )
+diff --git a/cat/BUILD.bazel b/cat/BUILD.bazel
+new file mode 100644
+index 00000000..3d9312d3
+--- /dev/null
++++ b/cat/BUILD.bazel
+@@ -0,0 +1,22 @@
++load("@rules_cc//cc:defs.bzl", "cc_binary")
++
++package(default_visibility = ["//visibility:public"])
++
++cc_binary(
++    name = "bsdcat",
++    srcs = [
++        "bsdcat.c",
++        "bsdcat.h",
++        "bsdcat_platform.h",
++        "cmdline.c",
++    ],
++    deps = [
++        "//libarchive",
++        "//libarchive_fe",
++    ],
++)
++
++alias(
++    name = "cat",
++    actual = ":bsdcat",
++)
+diff --git a/cpio/BUILD.bazel b/cpio/BUILD.bazel
+new file mode 100644
+index 00000000..5057343e
+--- /dev/null
++++ b/cpio/BUILD.bazel
+@@ -0,0 +1,30 @@
++load("@rules_cc//cc:defs.bzl", "cc_binary")
++
++package(default_visibility = ["//visibility:public"])
++
++SRCS = [
++    "cmdline.c",
++    "cpio.c",
++    "cpio.h",
++    "cpio_platform.h",
++]
++
++cc_binary(
++    name = "bsdcpio",
++    srcs = SRCS + select({
```

Comment:
nit: I would usually inline the `SRCS` here. I find it works a lot better with buildozer/etc

Reply:
I didn't realise it made working with buildozer worse. Thanks! It's updated now.